x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,12 +14,52 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  deleteDataFrameAnalyticsJobSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
+
+export interface DeleteDataFrameAnalyticsWithIndexStatus {
+  success: boolean;
+  error?: CustomHttpResponseOptions<ResponseError>;
+}
+
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteDestIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.deleteIndexPatternById(indexPatternId);
+}
 
 /**
  * Routes for the data frame analytics
  */
 export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitialization) {
+  async function userCanDeleteIndex(
+    context: RequestHandlerContext,
+    destinationIndex: string
+  ): Promise<boolean> {
+    if (!mlLicense.isSecurityEnabled()) {
+      return true;
+    }
+    const privilege = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', {
+      body: {
+        index: [
+          {
+            names: [destinationIndex], // uses wildcard
+            privileges: ['delete_index'],
+          },
+        ],
+      },
+    });
+    if (!privilege) {
+      return false;
+    }
+    return privilege.securityDisabled === true || privilege.has_all_requested === true;
```

Comment:
`securityDisabled` does not exist in `privilege`, it is only part of our client side `ml.hasPrivileges` function.